app/src/main/java/com/pvp/app/worker/WeeklyActivityWorkerSetup.kt

```diff
@@ -0,0 +1,39 @@
+package com.pvp.app.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.pvp.app.api.UserService
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import java.util.concurrent.TimeUnit
+
+@HiltWorker
+class WeeklyActivityWorkerSetup @AssistedInject constructor(
+    @Assisted context: Context,
+    @Assisted workerParams: WorkerParameters,
+    private val workManager: WorkManager
+) : CoroutineWorker(
+    context,
+    workerParams
+) {
+    override suspend fun doWork(): Result {
+        val requestPeriodic = PeriodicWorkRequestBuilder<WeeklyActivityWorker>(
+            repeatInterval = 7,
+            repeatIntervalTimeUnit = TimeUnit.DAYS
+        )
+            .build()
+
+        workManager.enqueueUniquePeriodicWork(
+            WeeklyActivityWorker.WORKER_NAME,
+            ExistingPeriodicWorkPolicy.CANCEL_AND_REENQUEUE,
+            requestPeriodic
+        )
+
+        return Result.success()
+    }
+}
```

Comment:
This _setup_ worker most likely can be extracted out of here into Application's periodic worker start method (your's), since all it does is just create periodic worker a minute after application installation. You probably could do a simple async coroutine delay before starting it (function will have to be suspendable and use `withContext` 🤔)

In my case, I used _setup_ worker because I had to delay the initialization of periodic worker for 0-23:59:59 interval (from least, to most possible time, theoretically). Delay can be assigned only to the OneTimeWorker, that's why another worker was required. Couldn't depend on coroutine delay, because the time is too great, it may not survive in memory, but workers persist in local storage (you could say).